client/packages/system/src/Item/context/useUnitVariant.ts

```diff
@@ -46,33 +55,64 @@ export const useInitUnitStore = () => {
   // TODO add user selected from app data
 };
 
-export const useUnitVariant = (itemId: string) => {
-  // TODO [state.items[itemId], state.userSelected[itemId]]
-  const item = useUnitStore(state => state.items[itemId], isEqual);
-  if (!item) {
+export const useUnitVariant = (
+  itemId: string
+): {
+  asPackUnit: (packSize: number) => string;
+  numberOfPacksFromQuantity: (totalQuantity: number) => number;
+  variantsControll?: {
+    variants: VariantNode[];
+    // Selected by user or mostUsed (calculated by backend)
+    activeVariant: VariantNode;
+    setUserSelectedVariant: (variantId: string) => void;
+  };
+} => {
+  const [item, userSelectedVariantId, setUserSelectedVariant] = useUnitStore(
+    state => [
+      state.items[itemId],
+      state.userSelectedVariants[itemId],
+      state.setUserSelectedVariant,
+    ],
+    isEqual
+  );
+
+  if (!item || item.variants.length == 0) {
     return {
-      asPackUnit: (packSize: number) => String(packSize),
-      numberOfPacksFromQuantity: (totalQuantity: number) => totalQuantity,
+      asPackUnit: packSize => String(packSize),
+      numberOfPacksFromQuantity: totalQuantity => totalQuantity,
     };
   }
 
-  const mostUsedVariant = item.variants.find(
-    ({ id }) => id === item.mostUsedVariantId
-  )?.packSize;
```

Comment:
previously mostUsedVariant was = packSize (as @mark-prins pointed out in previous review, i missed that)